### PR TITLE
Various tweaks to joins vignette

### PR DESCRIPTION
**Pull request by KyleHaynes**
Few tweaks to the new joins vignette.

- more consistent usage of full-stops.
- few typos.
- few grammatical improvements.

FYI @AngelFelizR (BTW - love the new vignette 🙏)

### CODE REVIEW

**Comment by AngelFelizR** [comment]
@KyleHaynes, thanks for taking the time to review the text.

**Comment by KyleHaynes** [comment]
No worries @AngelFelizR, feel free to object to any of my suggestions!

File: vignettes/datatable-joins.Rmd [hunk 1]
```diff
@@ -249,7 +249,7 @@ Products[
 ```
 
 
-##### Summarizing with on in data.table
+##### Summarizing with on in `data.table`
```
Comment (MichaelChirico) [inline vignettes/datatable-joins.Rmd:252]:
```suggestion
##### Summarizing with `on` in `data.table`
```

File: vignettes/datatable-joins.Rmd [hunk 2]
```diff
@@ -302,18 +302,18 @@ ProductReceived[Products,
                 nomatch = NULL]
 ```
 
-Despite both tables have the same information, they present some relevant differences:
+Despite both tables having the same information, they present some relevant differences:
```
Comment (MichaelChirico) [inline vignettes/datatable-joins.Rmd:305]:
```suggestion
Despite both tables having the same information, there are some relevant differences:
```

"they present" is redundant with the first bullet point

File: vignettes/datatable-joins.Rmd [hunk 3]
```diff
@@ -302,18 +302,18 @@ ProductReceived[Products,
...
 
-- They present different order for their columns
-- They have some name differences on their columns names:
-  - The `id` column of first table has the same information as the `product_id` in the second table.
-  - The `i.id` column of first table has the same information as the `id` in the second table.
+- They present different column ordering.
+- They have column name differences:
+  - The `id` column on the first table has the same information as the `product_id` in the second table.
+  - The `i.id` column on the first table has the same information as the `id` in the second table.
```
Comment (MichaelChirico) [inline vignettes/datatable-joins.Rmd:310]:
```suggestion
  - The `id` column in the first table has the same information as the `product_id` in the second table.
  - The `i.id` column in the first table has the same information as the `id` in the second table.
```

**Review by MichaelChirico — APPROVED** [review]
Thanks for taking the time to file!

**Pull request merged**